components/camel-elytron/src/main/java/org/apache/camel/component/elytron/ElytronConsumer.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.elytron;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import io.undertow.server.HttpServerExchange;
+import io.undertow.util.StatusCodes;
+import org.apache.camel.Processor;
+import org.apache.camel.component.undertow.UndertowConsumer;
+import org.apache.camel.component.undertow.UndertowEndpoint;
+import org.wildfly.security.auth.server.SecurityIdentity;
+import org.wildfly.security.authz.Roles;
+
+
+
+/**
+ * Consumer contains decides if request contains required roles (which are defined for endpoint)
+ *
+ * @author JiriOndrusek
+ */
+public class ElytronConsumer extends UndertowConsumer {
+
+    private Set<String> roles;
+    private List<String> allowedRoles;
+
+    public ElytronConsumer(UndertowEndpoint endpoint, Processor processor) {
+        super(endpoint, processor);
+    }
+
+    public ElytronEndpoint getElytronEndpoint() {
+        return (ElytronEndpoint) super.getEndpoint();
+    }
+
+    @Override
+    public void handleRequest(HttpServerExchange httpExchange) throws Exception {
+        SecurityIdentity identity = getElytronEndpoint().getElytronComponent().getSecurityDomain().getCurrentSecurityIdentity();
+
+        if (identity != null) {
+            //already authenticated
+            Set<String> roles = new HashSet<>();
+            Roles identityRoles = identity.getRoles();
+
+            if (identityRoles != null) {
+                for (String roleName : identityRoles) {
+                    roles.add(roleName);
+                }
+            }
+
+            if (isAllowed(roles, getElytronEndpoint().getAllowedRolesList())) {
+                super.handleRequest(httpExchange);
+            } else {
+                httpExchange.setStatusCode(StatusCodes.FORBIDDEN);
+            }
+
+            return;
```

Comment:
Please mind if I ask, but what would happen if the request reaches this line?

Reply:
It will return status code 403 and fails the exchange, You can see it in the junit test with bad role

Reply:
Ah right, didn't notice the that was setting `403` on the exchange. Thanks for the clarification!